cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java

```diff
@@ -139,6 +142,40 @@ private void initActionFailedExceptionConversion() {
                 .thenAnswer(invocation -> new Retry.ActionFailedException(new CloudConnectorException(invocation.getArgument(0, Exception.class))));
     }
 
+    private static Object[][] getDesResourceId() {
+        return new Object[][]{
+                {"/subscriptions/dummySubscriptionId/resourceGroups/dummyAlphaResourceGroup/providers/" +
+                        "Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaResourceGroup"},
+                {"/subscriptions/dummySubscriptionId/resourceGroups/dummyAlphaNumResourceGroup0910/providers/" +
+                        "Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaNumResourceGroup0910"},
+                {"/subscriptions/dummySubscriptionId/resourceGroups/dummyAlphaNumHyphenResourceGroup---0910/providers/" +
+                        "Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaNumHyphenResourceGroup---0910"},
+                {"/subscriptions/dummySubscriptionId/resourceGroups/dummyAlphaNumHyphenPeriodResourceGroup---....0910/providers/" +
+                        "Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaNumHyphenPeriodResourceGroup---....0910"},
+                {"/subscriptions/dummySubscriptionId/resourceGroups/dummyAlphaParenthesesPeriodResourceGroup---....)()(0910/providers/" +
+                        "Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaParenthesesPeriodResourceGroup---....)()(0910"},
+                {"/subscriptions/dummySubscriptionId/resourceGroups/dummyUnderscroceResourceGroup---___....)()(0910/providers/" +
+                        "Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyUnderscroceResourceGroup---___....)()(0910"}
+        };
+    }
+
+    @ParameterizedTest
+    @MethodSource("getDesResourceId")
+    public void testDiskEncryptionSetResourceGroupNameExtraction(String desResourceId, String resourceGroupName) {
+        String rgName = "";
+        Matcher matcher = underTest.DISK_ENCRYPTION_SET_RESOURCE_GROUP.matcher(desResourceId);
+        if (matcher.matches()) {
```

Comment:
We always expect the pattern to match, do not we? So I would add an `assertTrue(matcher.matches())` and eliminate the `if` condition entirely. Moreover, the dummy `rgName = ""` initialization is also unnecessary.